controllers/workspace/devworkspace_controller.go

```diff
@@ -232,16 +208,14 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 	routingStatus := provision.SyncRoutingToCluster(workspace, clusterAPI)
 	if !routingStatus.Continue {
 		if routingStatus.FailStartup {
-			reqLogger.Info("DevWorkspace start failed")
-			reconcileStatus.Phase = devworkspace.WorkspaceStatusFailed
 			// TODO: Propagate failure reason from devWorkspaceRouting
-			reconcileStatus.Conditions[devworkspace.WorkspaceFailedStart] = "Failed to install network objects required for devworkspace"
-			return reconcile.Result{}, routingStatus.Err
+			return r.failWorkspace("Failed to install network objects required for devworkspace", reqLogger, &reconcileStatus)
```

Comment:
Maybe it's a good time to fix TODO above?

Reply:
I think it's for a separate PR -- the underlying issue is that devworkspacerouting objects don't get a failure message, they just get marked as failed. The fix is to add a message field to the dwr spec and update the subcontroller to set it appropriately.

Reply:
Makes sense, there is a dedicated issue for that https://github.com/eclipse/che/issues/17455